glotaran/utils/io.py

```diff
@@ -294,3 +295,90 @@ def make_path_absolute_if_relative(path: Path) -> Path:
     if not path.is_absolute():
         path = get_script_dir(nesting=2) / path
     return path
+
+
+def create_clp_guide_dataset(
+    result: Result | xr.Dataset, clp_label: str, dataset_name: str | None = None
+) -> xr.Dataset:
+    """Create dataset for clp guidance.
+
+    Parameters
+    ----------
+    result: Result | xr.Dataset
+        Optimization result object or dataset, created with pyglotaran>=0.6.0.
+    clp_label : str
+        Label of the clp to guide.
+    dataset_name : str | None
+        Name of dataset to extract the guide from. Defaults to None.
+
+    Returns
+    -------
+    xr.Dataset
+        DataArray containing the clp guide, with ``clp_label`` dimension replaced by the
+        model dimensions first value.
+
+    Raises
+    ------
+    ValueError
+        If result is an instance of ``Result`` and ``dataset_name`` is ``None`` or not in result.
+    ValueError
+        If ``clp_labels`` is not in result.
+    ValueError
+        The the result dataset was created with pyglotaran<0.6.0.
+
+    Examples
+    --------
+    Extracting the clp guide from an optimization result object.
+
+    .. code-block:: python
+
+        from glotaran.io import save_dataset
+        from glotaran.utils.io import create_clp_guide_dataset
+
+        clp_guide = create_clp_guide_dataset(result, "species_1", "dataset_1")
+        save_dataset(clp_guide, "clp_guide__result_dataset_1__species_1.nc")
+
+    Extracting the clp guide from a result dataset loaded from file.
+
+    .. code-block:: python
+
+        from glotaran.io import load_dataset
+        from glotaran.io import save_dataset
+        from glotaran.utils.io import create_clp_guide_dataset
+
+        result_dataset = load_dataset("result_dataset_1.nc")
+        clp_guide = create_clp_guide_dataset(result_dataset, "species_1")
+        save_dataset(clp_guide, "clp_guide__result_dataset_1__species_1.nc")
+
+    """
+    if isinstance(result, xr.Dataset):
+        dataset = result
+    elif dataset_name is None or dataset_name not in result.data:
+        raise ValueError(
+            f"Unknown dataset {dataset_name!r}. "
+            f"Known datasets are:\n {list(result.data.keys())}"
+        )
+    else:
+        dataset = result.data[dataset_name]
+    if clp_label not in dataset.clp_label:
+        raise ValueError(
+            f"Unknown clp_label {clp_label!r}. "
+            f"Known clp_labels are:\n {list(dataset.clp_label.values)}"
+        )
+    if "model_dimension" not in dataset.attrs:
+        raise ValueError(
+            "Result dataset is missing attribute 'model_dimension', "
+            "which means that it was created with pyglotaran<0.6.0"
```

Comment:
Next word is capitalized, implying this is the end of the sentence so needs a period at the end.